sdk/formrecognizer/Azure.AI.FormRecognizer/src/AnalyzeResult.cs

```diff
@@ -11,7 +12,63 @@ namespace Azure.AI.FormRecognizer.DocumentAnalysis
     [CodeGenModel("AnalyzeResult")]
     public partial class AnalyzeResult
     {
+        internal AnalyzeResult(ApiVersion apiVersion, string modelId, StringIndexType stringIndexType, string content, IReadOnlyList<DocumentPage> pages, IReadOnlyList<DocumentTable> tables, IReadOnlyList<DocumentKeyValuePair> keyValuePairs, IReadOnlyList<DocumentEntity> entities, IReadOnlyList<DocumentStyle> styles, IReadOnlyList<AnalyzedDocument> documents)
+        {
+            ApiVersion = apiVersion;
+            ModelId = modelId;
+            StringIndexType = stringIndexType;
+            Content = content;
+            Pages = pages;
+            Tables = tables;
+            KeyValuePairs = keyValuePairs;
+            Entities = entities;
+            Styles = styles;
+            Documents = documents;
+
+            // Set page references for the 'Words' convenience properties.
+
+            foreach (DocumentPage page in Pages)
+            {
+                foreach (DocumentLine line in page.Lines)
+                {
+                    line.Page = page;
+                }
+            }
+
+            foreach (DocumentTable table in Tables)
+            {
+                table.Pages = Pages;
```

Comment:
Except for `DocumentLine`, all document elements can span across multiple pages, so we need to pass all analyzed pages.